### PR TITLE
feat: [PIE-3015]: variables view policy step

### PR DESCRIPTION
**Pull request by AshwinBhatkal-Harness**
##### Summary:

Variables view in policy step with expression and runtime support

##### Jira Links:

https://harness.atlassian.net/browse/PIE-3015

##### Screenshots:

Screenshot 2022-03-28 at 11 01 39 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by vkbansal — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/PolicyStep/utils.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export function getPolicySetValidationSchema({
   return Yup.mixed().test({
     test(value: string | string[]): boolean | Yup.ValidationError {
       if (Array.isArray(value)) {
-        if (value.length) {
+        if (value.length && !isEmpty(compact(value))) {
```
Comment (vkbansal) [inline src/modules/75-cd/components/PipelineSteps/PolicyStep/utils.ts:25]: Why this change is required?

Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/PipelineSteps/PolicyStep/utils.ts:25]: This is required to throw an error if there is an empty policy set expression list

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good
